### PR TITLE
 docs(cdk/testing): add guide for using harnesses

### PR DESCRIPTION
**Pull request by mmalerba**
related to #17118

### CODE REVIEW

**Comment by mmalerba** [comment]
I currently have this under `src/cdk/testing`, though it might make more sense to put it with the rest of the guides, WDYT?

**Comment by jelbourn** [comment]
We should have a standard overview for the `cdk/testing` stuff that documents the infrastructure parts, and then a separate guide for using the Angular Material harnesses

**Comment by mmalerba** [comment]
I'll create a separate PR for the overview, this one is more of a guide. I'll move the file to the same place as our other guides

**Comment by mmalerba** [comment]
Ok, ready for thorough review

**Comment by jelbourn** [comment]
Here's my commit with proposed edits:
https://github.com/jelbourn/components/commit/83a8a2b3477c6b014baa58d693ff0ae0e9498554

**Comment by mmalerba** [comment]
@jelbourn accepted your edits and added a few more of my own, PTAL

File: src/cdk/testing/using-component-harnesses.md [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+    fixture = TestBed.createComponent(IssueReportSelector);
+    fixture.detectChanges();
+    loader = TestbedHarnessEnvironment.loader(fixture);
+  });
+
+  it('should switch to bug report template', async () => {
+    expect(fixture.debugElement.query('bug-report-form')).toBeNull();
+    const select = await loader.getHarness(MatSelect);
+    await select.open();
+    const bugOption = await select.getOption({text: 'Bug'});
```
Comment (jelbourn) [inline src/cdk/testing/using-component-harnesses.md]: The select harness doesn't have a method like this now. Add it? We should add a `selectMatchingOption` method to the select harness to make this even more concise, _maybe_ also something like `openAndSelectMatchingOption`

Comment (mmalerba) [reply src/cdk/testing/using-component-harnesses.md]: I think @crisbeto is working on something like this in an open PR somewhere. But yeah, I do want to take a pass over all the harnesses before 9.0 and make sure they have the methods people need to do common actions quickly and easily

**Pull request merged**